Modified the record list to be reusable

Yqzzxlj opened the pull request.

Yqzzxlj committed:
Modified the record list.

Yqzzxlj commented:
![image](https://user-images.githubusercontent.com/43257534/57182253-aeb69e00-6ecf-11e9-8602-e7deb318fec5.png)
![image](https://user-images.githubusercontent.com/43257534/57182278-e1f92d00-6ecf-11e9-8b25-f4d326440957.png)
![image](https://user-images.githubusercontent.com/43257534/57182284-f0dfdf80-6ecf-11e9-851e-746d7cf50f99.png)

Time1ess commented on src/app/shared/components/shared-record-list/shared-record-list.component.ts:
```diff
@@ -0,0 +1,44 @@
+  styleUrls: ['./shared-record-list.component.css']
+})
+export class SharedRecordListComponent extends GenericListComponent<Record> {
+  @Input() recordListType: RecordListType;
```
Mark this as an optional field, `@Input() recordListType?: RecordListType`, and if this field is not defined, then in your `switch` we choose to fall back to retrieve all records.

Time1ess commented on src/app/shared/enums/record-list-type.enum.ts:
```diff
@@ -0,0 +1,6 @@
+/** Enums of record list types. */
+export enum RecordListType {
+    AllRecords = 0,
```
The enums should all be UPPER_CASE and separated by underscores.

Time1ess commented on src/app/shared/shared.module.ts:
```diff
@@ -87,6 +88,7 @@ import { OffCampusRecordDetailComponent } from './components/off-campus-record-d
     MatSnackBarModule,
 
     OffCampusRecordDetailComponent,
+    SharedRecordListComponent,
```
Since the component is declared in `SharedModule`, we don't need to import it, what `import` does is to import other modules and then we can use components, pipes or others `exported` from that module in our module. In your case, the `SharedRecordListComponent` is already in the `declarations` section, then all components in `SharedModule` will be able to use this component, and if you want other modules can use this component, just add it to `exports`, so other components in other modules can use your component as long as their modules have imported your `SharedModule`. Please also fix `OffCampusRecordDetailComponent` above this line.

Yqzzxlj committed:
1 Modified the recordListType to be optional
2 Change the enums to upper_case
3 import sharedModule rather than its components

Yqzzxlj committed:
fix code style

Time1ess commented on src/app/user/modules/records/components/record-detail/record-detail.component.spec.ts:
```diff
@@ -44,19 +35,11 @@ describe('RecordDetailComponent', () => {
-        FormsModule,
         NoopAnimationsModule,
         HttpClientTestingModule,
+        SharedModule,
```
I think you don't have to import `SharedModule`, instead, you can add the component you need in `declarations`.

Time1ess commented on src/app/user/modules/records/components/record-list/record-list.component.spec.ts:
```diff
@@ -1,33 +1,21 @@
-        MatIconModule,
-        MatProgressSpinnerModule,
-        MatPaginatorModule,
+        SharedModule,
```
Ditto.

Yqzzxlj committed:
modified the module in unit test
